src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1155,9 +1156,26 @@ private static FunctionResult ReadEnvironmentVariableResultBuilder(SemanticModel
                 }
                 else
                 {
+                    //log available environment variables if verbose logging is enabled
+                    if(model.Configuration.Analyzers.GetValue(LinterAnalyzer.LinterEnabledSetting, false) && model.Configuration.Analyzers.GetValue(LinterAnalyzer.LinterVerboseSetting, false)) {
+                        var availableEnvironmentVariables = model.Environment.GetVariableNames();
+                        diagnostics.Write(
+                            new Diagnostic(
+                                arguments[0].Span,
+                                DiagnosticLevel.Info,
+                                "Bicepparam ReadEnvironmentVariable function",
+                                "Available environment variables are: " + string.Join(", ", model.Environment.GetVariableNames()),
```

Comment:
Could this check do something similar to how we handle unknown properties in the type checker? There's a utility method to detect likely typos and include a "did you mean 'x'?" message on https://github.com/Azure/bicep/blob/126cd59e5ef58b1b55fbb087d0af7f3c2fe2176d/src/Bicep.Core/TypeSystem/TypeHelper.cs#L224

Reply:
That's a great idea!
I'll look into that.

Reply:
Done, let me know what you think.